napari/_qt/qt_viewer.py

```diff
@@ -385,6 +383,20 @@ def _create_canvas(self) -> None:
 
         self.canvas.destroyed.connect(self._diconnect_theme)
 
+    def _on_key_press(self, event):
+        """Handle key press events."""
+        if viewer := QtViewer.current_viewer():
+            viewer._key_map_handler.on_key_press(event)
+        else:
+            self._key_map_handler.on_key_press(event)
```

Comment:
As much as I love the walrus, I think it would be more readable like this:

```suggestion
        viewer = QtViewer.current_viewer() or self
        viewer._key_map_handler.on_key_press(event)
```